From 93674d494b23d952bfdb7aafecbf9a05f702da1b Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Sun, 25 Feb 2018 16:45:51 -0700
Subject: [PATCH 9/9] Revert "vfs,mm: fix a dead loop in
 truncate_inode_pages_range()"

This reverts commit c2a9737f45e27d8263ff9643f994bda9bac0b944.

This causes the kernel to not be able to read beyond 8TB on a single
device.
---
 mm/filemap.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/mm/filemap.c b/mm/filemap.c
index bf6aa30be58d..7807fa027973 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -2021,10 +2021,6 @@ static ssize_t generic_file_buffered_read(struct kiocb *iocb,
 	unsigned int prev_offset;
 	int error = 0;
 
-	if (unlikely(*ppos >= inode->i_sb->s_maxbytes))
-		return 0;
-	iov_iter_truncate(iter, inode->i_sb->s_maxbytes);
-
 	index = *ppos >> PAGE_SHIFT;
 	prev_index = ra->prev_pos >> PAGE_SHIFT;
 	prev_offset = ra->prev_pos & (PAGE_SIZE-1);
-- 
2.23.0

